docs/plugins/tutorials/simple-engine-plugins-ts.mdx

```diff
@@ -0,0 +1,536 @@
+tell the Hasura DDN Engine to utilize these URLs whenever a query hits the API. You can learn more about them
+[here](/supergraph-modeling/engine-plugins.mdx).
+
+### Step 14. Add the environment variables
+
+Since we've used the keys `PRE_PARSE_URL` and `PRE_RESPONSE_URL`, we'll need to map these values in our `.env` file in
+the root of our project.
+
+```yaml title="Add the following key-value pairs to your globals/metadata/subgraph.yaml file:"
+kind: Subgraph
+version: v2
+definition:
+  name: globals
+  generator:
+    rootPath: .
+  includePaths:
+    - metadata
+  #highlight-start
+  envMapping:
+    PRE_PARSE_URL:
+      fromEnv: PRE_PARSE_URL
+    PRE_RESPONSE_URL:
+      fromEnv: PRE_RESPONSE_URL
+  #highlight-end
+```
+
+```plaintext title="And these key-value pairs to your ddn directory's .env"
+PRE_PARSE_URL="http://local.hasura.dev:4000/pre-parse"
+PRE_RESPONSE_URL="http://local.hasura.dev:4000/pre-response"
+```
+
+### Step 15. Create a new build and test
+
+```sh title="Use the DDN CLI to create a new build:"
+ddn supergraph build local
+```
+
+```sh title="Then, kill your local services using CTRL+C before starting them back up:"
+ddn run docker-start
+```
+
+```graphql title="Finally, exeucte this query:"
+query GET_CUSTOMERS_AND_ORDERS {
+  customers {
+    id
+    name
+    email
+    orders {
+      id
+      orderDate
+      totalAmount
+    }
+  }
+}
+```
+
+```plaintext title="You'll see similar values logged to your server's terminal tab at the appropriate point in the query's execution:"
+This is running before the request is parsed! The user making this request is of role admin.
+
+
+This is running after the request is logged via the webhook before the response is sent to the client!
+
+
+    Response Data: {
+  "customers": [
+    {
+      "id": 1,
+      "name": "Alice",
+      "email": "alice@example.com",
+      "orders": [
+        {
+          "id": 1,
+          "orderDate": "2025-01-14",
+          "totalAmount": "99.99"
+        }
+      ]
+    },
+    {
+      "id": 2,
+      "name": "Bob",
+      "email": "bob@example.com",
+      "orders": [
+        {
+          "id": 2,
+          "orderDate": "2025-01-14",
+          "totalAmount": "49.50"
+        }
+      ]
+    },
+    {
+      "id": 3,
+      "name": "Charlie",
+      "email": "charlie@example.com",
+      "orders": [
+        {
+          "id": 3,
+          "orderDate": "2025-01-14",
+          "totalAmount": "75.00"
+        }
+      ]
+    }
+  ]
+}
+```
+
+## Next steps
+
+While this works locally, it's just as easy to also deploy this anywhere an Express application can be hosted. You can
+also use a separate context configuration for local vs. cloud development to test, collaborate, and deploy your DDN
+application — along with your plugins — whenever you wish. Learn more [here](/project-configuration/overview.mdx).
+
+Check out the other plugins we have available out-of-the-box in this directory, too!
```

Comment:
But we don't have any apart from this one right? Or are you referring to the allowlist plugin and the caching plugin?

Reply:
> Or are you referring to the allowlist plugin and the caching plugin?

Bingo! I didn't want to add a link here since that list will grow.